cmd/metacache-set.go

```diff
@@ -812,8 +812,8 @@ func listPathRaw(ctx context.Context, opts listPathRawOptions) (err error) {
 		return fmt.Errorf("listPathRaw: 0 drives provided")
 	}
 
-	// Disconnect from call above, but cancel on exit.
-	ctx, cancel := context.WithCancel(GlobalContext)
+	// Cancel on exit, also honor if caller cancels.
```

Comment:
Calls from `(*erasureObjects).listPath` must have this property, otherwise listing will stop/error as soon as the first batch of responses are returned.

Move this up there, then.

Reply:
@klauspost yes but why cache when listing itself gets canceled?  - I would prefer that listPathRaw() takes the right context based on the caller instead of it running forever without an explicit way to cancel the WalkDir() operations.

Reply:
@harshavardhana The context is canceled when the first call returns, when maxkeys is reached.

For these cases listing must continue, but for other callers not... so fix it in the caller.